### DISCUSSION TITLE
Docker 反向代理部署 发生客户端异常，部分 JS 文件有时会出现 514 错误响应

### DISCUSSION THREAD

**Discussion by fevrax** [lobehub/lobehub#2195] [Self Hosting] [answered]
### 💻 系统环境

Ubuntu

### 📦 部署环境

Docker

### 🌐 浏览器

Chrome

### 🐛 问题描述

Nginx 配置两个域名指向同一服务

```server
{
    listen 80;
	 listen 443 ssl http2;
        server_name gpt.fungs.cn chat.fungs.cn
        ......
```

访问 https://chat.fungs.cn/ 时，刷新多次后，部分 JS 文件有时会出现 514 错误响应。 而访问 gpt.fungs.cn 时，则一切运行正常。

GET https://chat.fungs.cn/_next/static/chunks/13695-40fba9481c53b990.js net::ERR_ABORTED 514

Uncaught (in promise) TypeError: Failed to register a ServiceWorker for scope ('https://chat.fungs.cn/') with script ('https://chat.fungs.cn/sw.js'): A bad HTTP response code (514) was received when fetching the script.

7528-2c20041b19b4c64b.js:1  ChunkLoadError: Loading chunk 13695 failed.
(timeout: https://chat.fungs.cn/_next/static/chunks/13695-40fba9481c53b990.js)
    at i.f.j (webpack-2d9df0573d13f1a3.js:1:18997)
    at webpack-2d9df0573d13f1a3.js:1:1502
    at Array.reduce (<anonymous>)
    at i.e (webpack-2d9df0573d13f1a3.js:1:1468)
    at 7528-2c20041b19b4c64b.js:1:106370
    at 7528-2c20041b19b4c64b.js:1:106572
    at t (7528-2c20041b19b4c64b.js:1:106775)
    
    f8b57a97-52d1c8984ef154e9.js:1  Uncaught Error: Minified React error #423; visit https://reactjs.org/docs/error-decoder.html?invariant=423 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at iB (f8b57a97-52d1c8984ef154e9.js:1:117726)
    at o4 (f8b57a97-52d1c8984ef154e9.js:1:94661)
    at f8b57a97-52d1c8984ef154e9.js:1:94483
    at o3 (f8b57a97-52d1c8984ef154e9.js:1:94490)
    at oQ (f8b57a97-52d1c8984ef154e9.js:1:91977)
    at oj (f8b57a97-52d1c8984ef154e9.js:1:91402)
    at MessagePort.T (7528-2c20041b19b4c64b.js:1:87710)

![image](https://github.com/lobehub/lobe-chat/assets/58434904/c9cc22a4-e707-4cdb-b1c6-60376b675be5)

### 🚦 期望结果

期望修复

### 📷 复现步骤

此域名 https://chat.fungs.cn 进入刷新几次

出错URL（可正常访问）：https://chat.fungs.cn/_next/static/chunks/app/chat/page-8199973102213c76.js

![image](https://github.com/lobehub/lobe-chat/assets/58434904/c39a2a3e-be61-4f4b-a4cb-23ef99056e97)

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
这个nginx配置问题感觉不是我们需要管的问题，自行找下解决方案？

[a comment by an automated account is not shown]

**Comment by fevrax** [comment]
好的，感谢。我也认为这可能是Nginx的问题。尽管Nginx配置相同，但更换为chat.*.*域名后就出现了问题。
后续在看看有没有其他人出现该问题。

[a comment by an automated account is not shown]

**Comment by peter5842** [comment]
我也遇到了这个问题，本来他运行的很好，现在就无法使用了。

[a comment by an automated account is not shown]

**Comment by fevrax** [comment]
> 我也遇到了这个问题，本来他运行的很好，现在就无法使用了。

我发现问题了。是我套了CDN取消就可以了

[a comment by an automated account is not shown]

**Comment by peter5842** [comment]
我发现好像是反代缓存的问题，我每清理一次就可以了？可能你的cdn也缓存了？

[a comment by an automated account is not shown]

**Comment by fevrax** [comment, marked as answer]
> 我发现好像是反代缓存的问题，我每清理一次就可以了？可能你的cdn也缓存了？

是的，一些文件获取不正确。把缓存、CND关了就好

[a comment by an automated account is not shown]
